pkg/controller/metrics/metrics.go

```diff
@@ -581,3 +589,23 @@ func GetClusterDeploymentType(obj metav1.Object) string {
 	}
 	return hivev1.DefaultClusterType
 }
+
+// ObserveControllerReconcileTime is used by all controllers in a defer block to log and report their elapsed time and
+// optionally an outcome in a consistent fashion.
+func ObserveControllerReconcileTime(controller string, start time.Time, outcome ReconcileOutcome, logger log.FieldLogger) {
+	dur := time.Since(start)
+	metricControllerReconcileTime.WithLabelValues(controller, string(outcome)).Observe(dur.Seconds())
+	fields := log.Fields{"elapsedMillis": dur.Milliseconds(), "outcome": outcome}
+	// Add a log field to categorize request duration into buckets. We can't easily query > in Kibana in
+	// OpenShift deployments due to logging limitations, so these constant buckets give us a small number of
+	// constant search strings we can use to identify slow reconcile loops.
+	for _, millis := range elapsedDurationMillisBuckets {
+		if dur.Milliseconds() >= millis {
+			fields["elapsedMillisGT"] = millis
+			break
+		}
+	}
+	logger.WithFields(fields).Info("reconcile complete")
+}
+
+var elapsedDurationMillisBuckets = []int64{120_000, 60_000, 30_000, 10_000, 5_000, 1_000, 0}
```

Comment:
I personally find it a little more readable if the buckets are defined using durations rather than milliseconds.
```suggestion
	for _, bucket := range elapsedDurationBuckets {
		if dur >= bucket {
			fields["elapsedMillisGT"] = bucket.Milliseconds()
			break
		}
	}
	logger.WithFields(fields).Info("reconcile complete")
}

var elapsedDurationBuckets = []time.Duration{2 * time.Minute, time.Minute, 30 * time.Second, 10 * time.Second, 5 * time.Second, time.Second, 0}
```

Reply:
Addressed